src/hotspot/share/cds/metaspaceShared.cpp

```diff
@@ -1319,21 +1317,18 @@ char* MetaspaceShared::reserve_address_space_for_archives(FileMapInfo* static_ma
       // via sequential file IO.
       address ccs_base = base_address + archive_space_size + gap_size;
       archive_space_rs = ReservedSpace(archive_space_size, archive_space_alignment,
-                                       os::vm_page_size(), (char*)base_address);
+                                       os::vm_page_size(), mtClassShared, (char*)base_address);
       class_space_rs   = ReservedSpace(class_space_size, class_space_alignment,
-                                       os::vm_page_size(), (char*)ccs_base);
+                                       os::vm_page_size(), mtClass, (char*)ccs_base);
     }
     if (!archive_space_rs.is_reserved() || !class_space_rs.is_reserved()) {
       release_reserved_spaces(total_space_rs, archive_space_rs, class_space_rs);
       return nullptr;
     }
-    // NMT: fix up the space tags
-    MemTracker::record_virtual_memory_type(archive_space_rs.base(), mtClassShared);
-    MemTracker::record_virtual_memory_type(class_space_rs.base(), mtClass);
```

Comment:
I assumed these (and others) were removed because the `MemTracker` updates had been pushed down into `ReserveSpace` itself, but I can't find them there - what am I missing?

Reply:
`archive_space_rs` and `class_space_rs` pass the MEMFLAGS to the `ReservedSpace` ctors a few lines above at 1272, 1319 and 1321.

Reply:
Yes I see the flags being passed to `ReservedSpace` but I don't see where `ReservedSpace` then calls `record_virtual_memory_type`.

Reply:
The `flag` is passed down in the call chain of `ReservedSpace ctor` -> `initialize()` -> `reserve()` -> `reserve_memory()` -> `os::xxx_reserve_memory_yyy()` where is passed to `MemTracker`. There will be no need to specifically call `MemTracker::record_virtual_memory_type(..., flag)` since the flag is already sent to `MemTracker` when  reserving the region.